src/Maker/MakeMigration.php

```diff
@@ -0,0 +1,142 @@
+use Symfony\Bundle\MakerBundle\ApplicationAwareMakerInterface;
+use Symfony\Bundle\MakerBundle\ConsoleStyle;
+use Symfony\Bundle\MakerBundle\DependencyBuilder;
+use Symfony\Bundle\MakerBundle\ExtraGenerationMakerInterface;
+use Symfony\Bundle\MakerBundle\InputConfiguration;
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\ArgvInput;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\BufferedOutput;
+
+/**
+ * @author Amrouche Hamza <hamza.simperfit@gmail.com>
+ * @author Ryan Weaver <ryan@knpuniversity.com>
+ */
+final class MakeMigration extends AbstractMaker implements ApplicationAwareMakerInterface, ExtraGenerationMakerInterface
+{
+    private $projectDir;
+
+    /**
+     * @var Application
+     */
+    private $application;
+
+    /**
+     * @var string
+     */
+    private $migrationOutput;
+
+    public function __construct(string $projectDir)
+    {
+        $this->projectDir = $projectDir;
+    }
+
+    public static function getCommandName(): string
+    {
+        return 'make:migration';
+    }
+
+    public function setApplication(Application $application)
+    {
+        $this->application = $application;
+    }
+
+    public function configureCommand(Command $command, InputConfiguration $inputConf)
+    {
+        $command
+            ->setDescription('Creates a new migration based on database changes.')
+            ->addOption('db', null, InputOption::VALUE_REQUIRED, 'The database connection name.')
+            ->addOption('em', null, InputOption::VALUE_OPTIONAL, 'The entity manager name')
+            ->addOption('shard', null, InputOption::VALUE_REQUIRED, 'The shard connection name.')
+            ->setHelp(file_get_contents(__DIR__.'/../Resources/help/MakeMigration.txt'))
+        ;
+    }
+
+    public function getParameters(InputInterface $input): array
+    {
+        return [
+            'options' => [
+                'db' => $input->getOption('db'),
+                'em' => $input->getOption('em'),
+                'shard' => $input->getOption('shard'),
+            ],
+        ];
+    }
+
+    public function getFiles(array $params): array
+    {
+        return [];
+    }
+
+    public function afterGenerate(ConsoleStyle $io, array $params)
+    {
+        $options = $params['options'];
+
+        $options['command'] = 'doctrine:migrations:diff';
+        $generateMigrationCommand = $this->application->find('doctrine:migrations:diff');
+
+        $commandOutput = new BufferedOutput($io->getVerbosity());
+        $generateMigrationCommand->run(new ArgvInput($options), $commandOutput);
+        $this->migrationOutput = $commandOutput->fetch();
+    }
+
+    public function writeSuccessMessage(array $params, ConsoleStyle $io)
+    {
+        if (false !== strpos($this->migrationOutput, 'No changes detected')) {
```

Comment:
I don't like this logic because it looks fragile ... but there's no other possible way to detect if a migration was generated or not. If MigrationsBundle supported JSON as one of its output formats ... but it doesn't.

Reply:
You're exactly right. So I'm depending on our tests to catch this :/